pkg/debugapi/pin.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package debugapi
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/gorilla/mux"
+)
+
+// pinChunk pin's the already created chunk given its address.
+// it fails if the chunk is not present in the local store.
+// It also increments a pin counter to keep track of how many pin requests
+// are originating for this chunk.
+func (s *server) pinChunk(w http.ResponseWriter, r *http.Request) {
+	addr, err := swarm.ParseHexAddress(mux.Vars(r)["address"])
+	if err != nil {
+		s.Logger.Debugf("debug api: parse chunk address: %v", err)
```

Comment:
Log messages in this methods should mention the handler that they originate from, like `s.Logger.Debugf("debug api: pin chunk: parse chunk address: %v", err)`. This goes for all log messages in debug api.